cupy/statistics/correlation.py

```diff
@@ -48,7 +59,83 @@ def corrcoef(a, y=None, rowvar=True, bias=None, ddof=None):
     return out
 
 
-# TODO(okuta): Implement correlate
+def correlate(a, v, mode='valid'):
+    """Returns the cross-correlation of two 1-dimensional sequences.
+
+    Args:
+        a (cupy.ndarray): first 1-dimensional input.
+        v (cupy.ndarray): second 1-dimensional input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        cupy.ndarray: Discrete cross-correlation of a and v.
+
+    .. seealso:: :func:`numpy.correlate`
+
+    """
+    if a.ndim != 1 or v.ndim != 1:
+        raise ValueError("object too deep for desired array")
+    conj_v = cupy.conj(v[::-1])
+    method = cupyx.scipy.signal.choose_conv_method(a, conj_v, mode)
+    if method == 'direct':
+        if cupy.iscomplexobj(v):
+            v = cupy.conj(v)
+        inverted, out = _dot_correlate(a, v, mode)
+        if inverted:
+            out = out[::-1]
+    elif method == 'fft':
+        out = cupy.math.misc._fftconvolve1d(a, conj_v, mode)
+    else:
+        raise ValueError('Unsupported method')
+    return out
+
+
+def _dot_correlate(a1, a2, mode):
```

Comment:
Why `dot_correlate` is placed in `cupy.statistics.correlation` while `_fftconvolve1d` is in `cupy.math.misc`? How about moving the implementations in `math.misc` with renaming to `_dot_convolve`?

Reply:
`dot_correlate` is placed  in `cupy.statistics.correlation` because `correlate` inputs are passed directly to it with reversing one of the inputs which is needed when used from `convolve`.
The same goes for `_fftconvolve1d`

Reply:
@asi1024 Should I still move `_dot_correlate` to `misc` as `_dot_convolve` ?